cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/AbstractDataprocProvisioner.java

```diff
@@ -127,10 +127,19 @@ public final ClusterStatus deleteClusterWithStatus(ProvisionerContext context, C
           // Status details is specific to dataproc jobs, so it was not added to RuntimeJobDetail spi.
           String statusDetails = ((DataprocRuntimeJobDetail) jobDetail).getJobStatusDetails();
           if (statusDetails != null) {
-            ProgramRunFailureException e = new ProgramRunFailureException(
-                String.format("Dataproc job '%s' status details: %s",
-                    ((DataprocRuntimeJobDetail) jobDetail).getJobId(), statusDetails));
-            LOG.error("Dataproc Job {}", jobDetail.getStatus(), e);
+            // Check if the failure is due to attempting to cancel a job already DONE
+            if (jobDetail.getStatus() == RuntimeJobStatus.FAILED && statusDetails.contains("is not supported in the current state: DONE")) {
```

Comment:
This exception of dataproc seems to be covered under `FAILED_PRECONDITION` 

and we are already handling it [DataprocRuntimeJobManager.java#L923](https://github.com/cdapio/cdap/blob/develop/cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/DataprocRuntimeJobManager.java#L923)

So, this check might not work. 

We are assuming Failure for all conditions of `FAILED_PRECONDITION` , may be we have a specific check there.